[vcpkg] Make configure meson sane and work for all targets.

Neumann-A opened the pull request.

Neumann-A committed:
fix configure meson for non windows builds.

Neumann-A committed:
change to the two ports using meson

Neumann-A committed:
fix passing of empty compiler arguments.

Neumann-A committed:
dont transform / into - for all flags

Neumann-A committed:
copy meson logs into the buildtree to better diagnose CI errors of meson ports

Neumann-A committed:
another change to the two ports to actually force a CI rebuild

Neumann-A committed:
add missing /pkgconfig dir to PKG_CONFIG_PATH

Voskrese commented:
test build `gobject-introspection`

not find CMake
> Found pkg-config: E:/tools/vcpkg/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe (0.29.2)
> Did not find CMake 'cmake'
> Found CMake: NO
> Run-time dependency glib-2.0 found: NO (tried pkgconfig and cmake)
> Looking for a fallback subproject for the dependency glib-2.0

add     

```
    vcpkg_find_acquire_program(CMAKE)
    get_filename_component(CMAKE_EXE_PATH ${CMAKE} DIRECTORY)
    vcpkg_add_to_path("${CMAKE_EXE_PATH}")
```

add `vcpkg_find_acquire_program.cmake`
```
  elseif(VAR MATCHES "CMAKE")
    set(PROGNAME cmake)
    set(PATHS "${PROGRAM_FILES_PLATFORM_BITNESS}/CMake/bin" "${PROGRAM_FILES_32_BIT}/CMake/bin" "${DOWNLOADS}/tools/CMake/bin")
```

:((
glib-2.0 **found: NO** (tried pkgconfig and **cmake)**
```
Found pkg-config: E:/tools/vcpkg/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe (0.29.2)
Found CMake: C:/Program Files/CMake/bin\cmake.EXE (3.16.4)
Run-time dependency Found pkg-config: E:/tools/vcpkg/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe (0.29.2)
Found CMake: C:/Program Files/CMake/bin\cmake.EXE (3.16.4)
Run-time dependency glib-2.0 found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency glib-2.0

meson.build:134:0: ERROR: Automatic wrap-based subproject downloading is disabled
Looking for a fallback subproject for the dependency glib-2.0

meson.build:134:0: ERROR: Automatic wrap-based subproject downloading is disabled
```

[meson-log.txt](https://github.com/microsoft/vcpkg/files/4333792/meson-log.txt)

Voskrese commented:
test add `glib-2.0.pc` to **debug** `vcpkg\installed\x64-windows\debug\lib\pkgconfig`

```
prefix=../..
exec_prefix=${prefix}
libdir=${prefix}/lib
bindir=${prefix}/bin
datarootdir=${prefix}/share
datadir=${datarootdir}
toolsdir=${prefix}/bin
includedir=${prefix}/../include

glib_genmarshal=${bindir}/glib-genmarshal
gobject_query=${bindir}/gobject-query
glib_mkenums=${bindir}/glib-mkenums

Name: GLib
Description: C Utility Library
Version: 2.60.4
Libs: -lintl -L${libdir} -lglib-2.0
Libs.private: -lws2_32 -lwinmm
Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include
```

not work **msys pkg-config** `msys2/msys64/usr/bin/pkg-config.exe`
path **e:/**
```
set PKG_CONFIG_PATH=E:/tools/vcpkg/installed/x64-windows/debug/lib/pkgconfig
E:/tools/vcpkg/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe --define-variable=prefix=E:/tools/vcpkg/installed/x64-windows/debug --modversion glib-2.0
```

> Package glib-2.0 was not found in the pkg-config search path.
> Perhaps you should add the directory containing `glib-2.0.pc'
> to the PKG_CONFIG_PATH environment variable
> No package 'glib-2.0' found

work path **/e/...**
```
set PKG_CONFIG_PATH=/e/tools/vcpkg/installed/x64-windows/debug/lib/pkgconfig
E:/tools/vcpkg/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe --define-variable=prefix=/e/tools/vcpkg/installed/x64-windows/debug --modversion glib-2.0
```

> 2.60.4

Voskrese commented:
work windows https://github.com/pkgconf/pkgconf

```
vcpkg_download_distfile(ARCHIVE
    URLS "https://github.com/pkgconf/pkgconf/archive/pkgconf-1.6.3.tar.gz"
    FILENAME "pkgconf-1.6.3.tar.gz"
    SHA512 e5550b68939d6936afa957b0a0fb4bcb36731ad6b617199a720ce9d3f0e8a93c12df18a8192b3e7dff43dbb4c9a9ba7973e97c849d1f84653a4fb40ce20c6eb2
)

vcpkg_extract_source_archive_ex(
    OUT_SOURCE_PATH SOURCE_PATH
    ARCHIVE ${ARCHIVE}
)

vcpkg_configure_meson(
    SOURCE_PATH ${SOURCE_PATH}
    OPTIONS
      --backend=ninja
	  -Dtests=false
)

vcpkg_install_meson()
```

```
set PKG_CONFIG_PATH=E:/tools/vcpkg/installed/x64-windows/debug/lib/pkgconfig
pkgconf.exe --define-variable=prefix=E:/tools/vcpkg/installed/x64-windows/debug --modversion glib-2.0
```

> 2.60.4

Neumann-A commented:
Maybe need to try to switch to the mingw pkg-config paket then or do the path conversion manually. For some reason it worked for me without that.

Voskrese commented:
there may be problems on my side, in variables, but I couldn't track the error

> we need to test it, and we need user reviews

Neumann-A commented:
I think the problem is that i pass a Cmake path here and in the x window PR do a conversion to native path. The cmake path might screw up the automatic path conversion

Voskrese commented:
if comment, then something turns out
```
#        if(NOT PKGCONFIG MATCHES "--define-variable=prefix")
#            set(ENV{PKG_CONFIG} "${PKGCONFIG} --define-variable=prefix=${CURRENT_INSTALLED_DIR}")
#        endif()
```

Neumann-A commented:
If you comment that out you use the automatic prefix from pkgconfig on windows. This does not work if prefix is removed from *.pc file has mentioned in the vcpkg_fixup_pkgconfig PR. 
can you post the `gobject-introspection` port folder for testing?

Voskrese commented:
>can you post the gobject-introspection port folder for testing?
I took the first packet of gnome, which is not in vcpkg

Voskrese commented:
```
vcpkg_download_distfile(ARCHIVE
    URLS "https://gitlab.gnome.org/GNOME/gobject-introspection/-/archive/1.64.0/gobject-introspection-1.64.0.zip"
    FILENAME "gobject-introspection-1.64.0.zip"
    SHA512 b84398a73e594a8b846dbd2a863feddff2007fba3c7fd073dd4a961b10ca30d5d23a26ddcaaf227f83cf49f65b9eb20fee806e22348610dfc14916bb74269dfd
)

vcpkg_extract_source_archive_ex(
    OUT_SOURCE_PATH SOURCE_PATH
    ARCHIVE ${ARCHIVE}
)

vcpkg_find_acquire_program(FLEX)
vcpkg_find_acquire_program(BISON)
vcpkg_find_acquire_program(DOXYGEN)
vcpkg_find_acquire_program(PYTHON3)

get_filename_component(PYTHON3_DIR "${PYTHON3}" DIRECTORY)
get_filename_component(FLEX_DIR "${FLEX}" DIRECTORY)
get_filename_component(BISON_DIR "${BISON}" DIRECTORY)
get_filename_component(DOXYGEN_DIR "${DOXYGEN}" DIRECTORY)

set(ENV{PATH} ";$ENV{PATH};${FLEX_DIR};${BISON_DIR};${DOXYGEN_DIR};${PYTHON3_DIR}")

vcpkg_configure_meson(
	SOURCE_PATH ${SOURCE_PATH}
    OPTIONS
		--backend=ninja
		-Dcairo_libname=cairo-gobject.dll
		-Dgtk_doc=false
		-Ddoctool=disabled
		-Dpython=${PYTHON3}
)
vcpkg_install_meson()
vcpkg_copy_pdbs()
file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/gobject-introspection RENAME copyright)

```

Neumann-A committed:
fix pkgconfig setup

Neumann-A commented:
@Voskrese: Could you try out the latest commit? There seems to be some hidden path conversions going on because meson tells me glib has been found but if I try to copy the command meson uses int a shell it tells me it cannot find it.

Voskrese commented:
> I'll try the changes later.
> there is another option from MESON

add `-Dpkg_config_path`

```
    list(APPEND _vcm_OPTIONS_DEBUG 
	    --prefix ${CURRENT_PACKAGES_DIR}/debug 
	    --includedir ../include
	    -Dpkg_config_path=${CURRENT_INSTALLED_DIR}/debug/lib/pkgconfig,${CURRENT_INSTALLED_DIR}/debug/share/pkgconfig
    list(APPEND _vcm_OPTIONS_RELEASE
	    --prefix  ${CURRENT_PACKAGES_DIR}
	    -Dpkg_config_path=${CURRENT_INSTALLED_DIR}/lib/pkgconfig,${CURRENT_INSTALLED_DIR}/share/pkgconfig
	  )
```

Voskrese commented:
module `CMake\share\cmake-3.15\Modules\FindPkgConfig.cmake`
add     `include(FindPkgConfig)` ?

```
    find_program(PKGCONFIG pkg-config REQUIRED)
    include(FindPkgConfig)
    else()
        debug_message(STATUS "PKG_CONFIG found in ENV! Using $ENV{PKG_CONFIG}")
        set(PKGCONFIG $ENV{PKG_CONFIG})
    endif()
```

Voskrese commented:
@Neumann-A  patch ?

# fix the PKG_CONFIG_PATH transformed into a cmake path with incorrect msys pathes
https://github.com/videolan/vlc/blob/75cd97f6b1659e0dea16f18ef073f34bafe67911/extras/tools/cmake-msys-FindPkg.patch#L7

Voskrese commented:
hmm... work `MSYS` PKGCONFIG

add macros

```
macro(CONVERT_CYGWIN_PATH _path)
  if (WIN32)
    if( NOT DEFINED CYGPATH_EXECUTABLE OR NOT EXISTS ${CYGPATH_EXECUTABLE} )
	  find_program(CYGPATH_EXECUTABLE NAMES cygpath)
	endif()
	  message(STATUS "convert input")
	  message(${${_path}})
	  string (REGEX REPLACE "\\?space\\?" " " ${_path} ${${_path}}) 
	  message(STATUS "replace spaces")
	  message(${${_path}})
      EXECUTE_PROCESS(COMMAND ${CYGPATH_EXECUTABLE} -up "${${_path}}"
        OUTPUT_VARIABLE ${_path})
      string (STRIP ${${_path}} ${_path})
	  string(REGEX REPLACE "\\\\" "/" ${_path} ${${_path}}) 
	  message(STATUS "convert output")
	  message(${${_path}})
  endif (WIN32)
endmacro(CONVERT_CYGWIN_PATH)
```

```
    # configure release
    if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
        message(STATUS "Configuring ${TARGET_TRIPLET}-rel")
        file(MAKE_DIRECTORY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel)        
        #setting up PKGCONFIG
        if(NOT PKGCONFIG MATCHES "--define-variable=prefix")
            file(TO_CMAKE_PATH ${CURRENT_INSTALLED_DIR} PKGCONFIG_PREFIX)
	    CONVERT_CYGWIN_PATH(PKGCONFIG_PREFIX) ### <<-- add convert
            set(ENV{PKG_CONFIG} "${PKGCONFIG} --define-variable=prefix=${PKGCONFIG_PREFIX}")
        endif()
		
        message("Converted \"${CURRENT_INSTALLED_DIR}\" to \"${PKGCONFIG_PREFIX}\".")
 
        file(TO_CMAKE_PATH "${CURRENT_INSTALLED_DIR}/lib/pkgconfig" PKGCONFIG_INSTALLED_DIR)
	CONVERT_CYGWIN_PATH(PKGCONFIG_INSTALLED_DIR) ### <<-- add convert
        if(ENV{PKG_CONFIG_PATH})
            set(BACKUP_ENV_PKG_CONFIG_PATH_RELEASE $ENV{PKG_CONFIG_PATH})
			set(ENV{PKG_CONFIG_PATH} "${PKGCONFIG_INSTALLED_DIR}${VCPKG_HOST_PATH_SEPARATOR}$ENV{PKG_CONFIG_PATH}")
        else()
            set(ENV{PKG_CONFIG_PATH} "${PKGCONFIG_INSTALLED_DIR}")
        endif()
```


```
-- Acquiring MSYS Packages...
-- Acquiring MSYS Packages... OK
-- Found PkgConfig: E:/tools/vcpkg/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe (found version "0.29.2")
-- Configuring x64-windows-rel
-- convert input
E:/tools/vcpkg/installed/x64-windows
-- replace spaces
E:/tools/vcpkg/installed/x64-windows
-- convert output
/e/tools/vcpkg/installed/x64-windows
Converted "E:/tools/vcpkg/installed/x64-windows" to "/e/tools/vcpkg/installed/x64-windows".
-- convert input
E:/tools/vcpkg/installed/x64-windows/lib/pkgconfig
-- replace spaces
E:/tools/vcpkg/installed/x64-windows/lib/pkgconfig
-- convert output
/e/tools/vcpkg/installed/x64-windows/lib/pkgconfig
-- Configuring x64-windows-rel
```

Voskrese commented:
or no CYGWIN

```
function(CONVERT_CYGWIN_PATH PATH RESULT)
    file(TO_CMAKE_PATH ${PATH} CP)   	# changes \ to / on windows
    string(REGEX REPLACE "^([a-zA-Z]):/" "/\\1/" CP2 ${CP})   	# c:/ -> /c/
    set(${RESULT} "${CP2}" PARENT_SCOPE)   	# or set(${RESULT} "/cygdrive${CP2}" PARENT_SCOPE) ### /c/ -> /cygdrive/c/
endfunction()
```

~~file(TO_CMAKE_PATH ${CURRENT_INSTALLED_DIR} PKGCONFIG_PREFIX)~~
`CONVERT_CYGWIN_PATH(${CURRENT_INSTALLED_DIR} PKGCONFIG_PREFIX)`

and

~~file(TO_CMAKE_PATH "${CURRENT_INSTALLED_DIR}/lib/pkgconfig" PKGCONFIG_INSTALLED_DIR)~~
`CONVERT_CYGWIN_PATH("${CURRENT_INSTALLED_DIR}/lib/pkgconfig" PKGCONFIG_INSTALLED_DIR)`

Neumann-A commented:
>add -Dpkg_config_path ///not work msys pkg-config

Probably needs the same handling as the compiler args as the paths must be passed as ['path1', 'path2'] etc

>include(FindPkgConfig)

Is overkill for our usage. Pulls in too many unnecessary macros. We just need the EXECUTABLE_PATH and that is it.

> hmm... work MSYS PKGCONFIG

did you also try out a clean checkout of f7d3858 ? For me this worked and I would like to not add in any strange kind of path conversions because it just complicates the code. If it does not work we could try to switch to `mingw-w64-pkg-config` instead.

Voskrese commented:
> did you also try out a clean checkout of f7d3858 ?
no, it doesn't work, the paths are not acceptable as **E:/ or E:\\** (C:/ or C:\\)
only works with **/E/**

> I tested on one laptop, maybe as I wrote above, the other will work
> 
> we need user reviews

Neumann-A commented:
@Voskrese 
Strange, on my clean laptop I get the following output
[meson-log.txt](https://github.com/microsoft/vcpkg/files/4337116/meson-log.txt)
`Run-time dependency glib-2.0 found: YES 2.60.4`
Pkgfile:
[glib-2.0.pc.txt](https://github.com/microsoft/vcpkg/files/4337123/glib-2.0.pc.txt)

Command Line Output:
```
C:\vcpkg_make>set BAK_PATH=%PATH%
C:\vcpkg_make>set PATH=%BAK_PATH%;C:\vcpkg_make\downloads\tools\msys2\msys64\usr\bin
C:\vcpkg_make>set PKG_CONFIG_PATH=C:\vcpkg_make\installed\x86-windows\debug\lib\pkgconfig
C:\vcpkg_make>C:/vcpkg_make/downloads/tools/msys2/msys64/usr/bin/pkg-config.exe --define-variable=prefix=C:\vcpkg_make\installed\x86-windows\debug --modversion glib-2.0
2.60.4
```

Voskrese commented:
I tried many ways and revised the variables
all the same

in general, I do not know what the reason is, if everything works, then you need to combine/merge

Neumann-A commented:
@Voskrese: What shell are you running the commands in? Maybe that is creating the issue?

Voskrese commented:
@Neumann-A 
cleaned the variables
everything works with **TO_NATIVE_PATH**

Voskrese commented:
@dan-shaw 
merge, we need to move on, a lot of new ports and updates on MESON from GNOME

Neumann-A committed:
add share/pkgconfig to PKG_CONFIG_PATH

ras0219-msft commented:
/azp run

Voskrese commented on scripts/cmake/vcpkg_configure_meson.cmake:
```diff
@@ -38,34 +38,45 @@ function(vcpkg_configure_meson)
+    string(APPEND MESON_RELEASE_LDFLAGS " ${CMAKE_SHARED_LINKER_FLAGS_RELEASE${MESON_CMAKE_FLAG_SUFFIX}}")
     
     # select meson cmd-line options
     list(APPEND _vcm_OPTIONS -Dcmake_prefix_path=${CURRENT_INSTALLED_DIR})
```
```suggestion
    list(APPEND _vcm_OPTIONS "-Dcmake_prefix_path=['${CURRENT_INSTALLED_DIR}','${CURRENT_INSTALLED_DIR}/share']" #work windows search in folder share module cmake
```

Voskrese replied:
```
Called `C:/Program Files/CMake/bin\cmake.EXE --trace-expand -DNAME=libffi -DARCHS= -DCMAKE_PREFIX_PATH=E:/tools/vcpkg/installed/x64-windows;E:/tools/vcpkg/installed/x64-windows/share .` in E:\tools\vcpkg\buildtrees\glib\x64-windows-rel\meson-private\cmake_libffi -> 0
Guessed CMake target 'libffi'
CMake TARGET:
  -- name:      libffi
  -- type:      STATIC
  -- imported:  True
  -- properties: {
      'IMPORTED_CONFIGURATIONS': ['DEBUG', 'RELEASE']
      'IMPORTED_LOCATION_DEBUG': ['E:/tools/vcpkg/installed/x64-windows/debug/lib/libffi.lib']
      'IMPORTED_LOCATION_RELEASE': ['E:/tools/vcpkg/installed/x64-windows/lib/libffi.lib']
     }
  -- tline: CMake TRACE: E:/tools/vcpkg/installed/x64-windows/share/libffi/libffiTargets.cmake:53 add_library(['libffi', 'STATIC', 'IMPORTED'])
Include Dirs:         []
Compiler Definitions: []
Compiler Options:     []
Libraries:            ['E:/tools/vcpkg/installed/x64-windows/lib/libffi.lib']
Dependency libffi found: NO unknown version, but need: ['>= 3.0.0']
Run-time dependency libffi found: YES
```

Voskrese replied:
https://cmake.org/cmake/help/latest/command/find_package.html#id5

ONLY Unix

```
<prefix>/(lib/<arch>|lib*|share)/cmake/<name>*/                 (U)
<prefix>/(lib/<arch>|lib*|share)/<name>*/                       (U)
<prefix>/(lib/<arch>|lib*|share)/<name>*/(cmake|CMake)/         (U)
```

Neumann-A replied:
I that would be the case all `find_package` calls in vcpkg would fail. (see -> vcpkg.cmake)
I think CMakes docs are wrong here.

Voskrese replied:
this is the only simple workaround, but a working one, this is as an example.
we need to add the code to make it work

Voskrese commented on scripts/cmake/vcpkg_configure_meson.cmake:
```diff
@@ -76,45 +87,149 @@ function(vcpkg_configure_meson)
     list(APPEND _vcm_OPTIONS_RELEASE --prefix  ${CURRENT_PACKAGES_DIR})
     
     vcpkg_find_acquire_program(MESON)
+    
```
get_filename_component(CMAKE_PATH ${CMAKE_COMMAND} DIRECTORY)

Voskrese commented on scripts/cmake/vcpkg_configure_meson.cmake:
```diff
@@ -76,45 +87,149 @@ function(vcpkg_configure_meson)
-    if(CMAKE_HOST_WIN32)
-        set(_PATHSEP ";")
+    vcpkg_add_to_path("${NINJA_PATH}")
+
```
vcpkg_add_to_path("${CMAKE_PATH}")

Voskrese commented:
`${VCPKG_HOST_PATH_SEPARATOR}` not work windows host msys, windows `;` , **msys** `:`

work     
```
    if(CMAKE_HOST_WIN32)
        set(_SEP ";")
    else()
        set(_SEP ":")
    endif()
```

dan-shaw commented:
/azp run

Neumann-A committed:
Add CMake path to PATH for meson

Neumann-A committed:
add share folder to prefix for meson

Neumann-A committed:
remove double message

Voskrese commented:
@Neumann-A 
it is better to wait with this, the scripts do not correctly interpret the paths to the libraries, and this causes a lot of errors

only windows
"-D **cmake_prefix_path** =['${.....}','${ **CURRENT_INSTALLED_DIR}/share** ']")

JackBoosY commented:
/azp run

Neumann-A committed:
add libdir so that meson does not install into architecture folder in linux
and make prefix path target dependent. (Or should it be host dependent?)

JackBoosY reviewed (changes requested):
We should update `fribidi` and `libepoxy` version in _CONTROL_ file.

Neumann-A committed:
bump controls

dan-shaw commented on scripts/cmake/vcpkg_configure_meson.cmake:
```diff
@@ -38,83 +38,204 @@ function(vcpkg_configure_meson)
+    #Extract compiler flags
+    if(NOT VCPKG_CHAINLOAD_TOOLCHAIN_FILE)
+        set(MESON_CMAKE_FLAG_SUFFIX "_INIT")
+        if(NOT DEFINED VCPKG_CMAKE_SYSTEM_NAME OR _TARGETTING_UWP)
```
Would `VCPKG_TARGET_IS_WINDOWS` work?

Neumann-A replied:
updated was copied from `vcpkg_configure_cmake`. 
Nevertheless, another approach will be needed to obtain the correct set of compiler flags e.g. a call to a dummy cmake project to extract the flags since CMake will possibly add flags to _INIT variables. As such vcpkg only controls compiler flags for windows while all other platforms leverage CMake's defaults.

dan-shaw commented on scripts/cmake/vcpkg_configure_meson.cmake:
```diff
@@ -38,83 +38,204 @@ function(vcpkg_configure_meson)
     # select meson cmd-line options
-    list(APPEND _vcm_OPTIONS -Dcmake_prefix_path=${CURRENT_INSTALLED_DIR})
+    if(VCPKG_TARGET_IS_WINDOWS)
+        list(APPEND _vcm_OPTIONS "-Dcmake_prefix_path=['${CURRENT_INSTALLED_DIR}','${CURRENT_INSTALLED_DIR}/share']")
```
Why do we need the `share` directory on `cmake_prefix_path`? Will `share` get searched from`${CURRENT_INSTALLED_DIR}`?

dan-shaw replied:
Seems that `meson` doesn't follow the CMake search rules when using `cmake_prefix_path`. Another question: why don't we add `share` as a path to non-windows targets?

Neumann-A replied:
see https://github.com/microsoft/vcpkg/pull/10395#discussion_r395254705

dan-shaw commented:
Since we are adding support for Linux, could we recommend the user to use `apt-get` and `pip` to install meson (or what is officially supported) in `vcpkg_acquire_program`? Meanwhile, we will add meson for Linux to the CI.

Neumann-A commented:
@dan-shaw 
```
      elseif(DEFINED APT_PACKAGE_NAME AND CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
        set(EXAMPLE ":\n    sudo apt-get install ${APT_PACKAGE_NAME}")
      endif()
```
So Linux user will get a message if meson is not installed

Neumann-A committed:
update toolchain selector to use VCPKG_TARGET_IS_<paltform>.

JackBoosY commented:
The osg-qt regression is not related to this PR.

dan-shaw commented:
/azp run

dan-shaw commented:
/azp run

Neumann-A committed:
change pkgconfig prefix and path.....
meson is not correctly handling native windows paths and eating "\"
PKG_CONFIG_PATH also nees : to correctly separate paths
Prefix still must be near to native path but with / instead of \ C:/somepath/

JackBoosY commented:
Needs to wait to resolve the osg-qt regression in #10637.

JackBoosY commented:
/azp run

dan-shaw commented:
@Neumann-A thanks for the PR!
